tests/test_impls.py

```diff
@@ -50,18 +62,214 @@ def check_initial_contents(filename):
 
 
 @pytest.mark.parametrize(
-    ["initial_req", "cuda_suffix", "cuda_major", "expected_req"],
     [
-        ("cupy", "", "11", "cupy-cuda11x"),
-        ("cupy", "", "12", "cupy-cuda12x"),
-        ("cupy", "", None, "cupy"),
-        *((name, "-cu11", "11", f"{name}-cu11") for name in _VERSIONED_RAPIDS_WHEELS),
-        *((name, "-cu12", "12", f"{name}-cu12") for name in _VERSIONED_RAPIDS_WHEELS),
-        *((name, "", None, name) for name in _VERSIONED_RAPIDS_WHEELS),
-        ("poetry", "-cu11", "11", "poetry"),
-        ("poetry", "-cu12", "12", "poetry"),
-        ("poetry", "", None, "poetry"),
+        "pyproject_dir",
+        "dependencies_file",
+        "write_dependencies_file",
+        "cuda_version",
+        "cuda_suffix",
+        "cuda_python_requirement",
+        "matrix",
+        "arch_requirement",
+    ],
+    [
+        (
+            ".",
+            "dependencies.yaml",
+            True,
+            ("11", "5"),
+            "-cu11",
+            "cuda-python>=11.5,<11.6.dev0",
+            "",
+            "some-x86-package",
+        ),
+        (
+            ".",
+            "dependencies.yaml",
+            True,
+            ("11", "5"),
+            "-cu11",
+            "cuda-python>=11.5,<11.6.dev0",
+            "arch=aarch64",
+            "some-arm-package",
+        ),
+        (
+            "python",
+            "../dependencies.yaml",
+            True,
+            ("12", "1"),
+            "-cu12",
+            "cuda-python>=12.1,<12.2.dev0",
+            "arch=ppc64",
+            "some-ppc-package",
```

Comment:
We don't support ppc64le in RAPIDS (if we did, we would probably spell it `ppc64le` instead of `ppc64`). Let's avoid using this as a test case.